Firmware/Marlin_main.cpp

```diff
@@ -4027,6 +4027,8 @@ static void extended_capabilities_report()
 #endif //FANCHECK and TACH_0 or TACH_1
     // AUTOREPORT_POSITION (M114)
     cap_line(PSTR("AUTOREPORT_POSITION"), ENABLED(AUTO_REPORT));
+    // M108, M112, M410 and Prusa specific M603.
+    cap_line(PSTR("EMERGENCY_PARSER"), 1);
```

Comment:
@leptun would this be sufficient to expose EMERGENCY_PARSER?

Reply:
Hmm... I'm not sure. I don't think it's enough since afaik we don't constantly run get_command()

Reply:
Expose where / for what?

```
2022-08-28 15:35:13,239 - Recv: FIRMWARE_NAME:Prusa-Firmware 3.11.1 based on Marlin FIRMWARE_URL:https://github.com/prusa3d/Prusa-Firmware PROTOCOL_VERSION:1.0 MACHINE_TYPE:Prusa i3 MK3S EXTRUDER_COUNT:1 UUID:00000000-0000-0000-0000-000000000000
2022-08-28 15:35:13,243 - Recv: Cap:AUTOREPORT_TEMP:1
2022-08-28 15:35:13,270 - Recv: Cap:AUTOREPORT_FANS:1
2022-08-28 15:35:13,362 - Recv: Cap:AUTOREPORT_POSITION:1
2022-08-28 15:35:13,451 - Recv: Cap:EMERGENCY_PARSER:1
2022-08-28 15:35:13,565 - Recv: Cap:EXTENDED_M20:1
2022-08-28 15:35:13,586 - Recv: Cap:PRUSA_MMU2:1
```

OctoPrint picks that up.

```
2022-08-28 00:59:42,383 - octoprint.util.comm - INFO - Printer reports firmware name "Prusa-Firmware 3.11.1 based on Marlin"
2022-08-28 00:59:42,383 - octoprint.util.comm - INFO - Detected Prusa firmware, enabling relevant features for issue free communication
2022-08-28 00:59:42,413 - octoprint.util.comm - INFO - Firmware states that it supports temperature autoreporting
2022-08-28 00:59:42,446 - octoprint.util.comm - INFO - Firmware states that it supports emergency GCODEs to be sent without waiting for an acknowledgement first
```
(last line)

Reply:
At least I interpreted the question as "are these changes enough to make the emergency parser feature to work?". Because, we can report that we support it, but that doesn't mean the commands get queued properly. I hope they do, but I think they don't

Reply:
Ahh, ok.

From my observation there are some delays like when "G28 W" is being made for example. When actual object printing object it was reacting quite fast (but obviously have no idea about possible corner cases).

Question is can it be made more responsive (without some huge rewrite) or are such delays acceptable for this capability.

Reply:
Marlin did that via state machine in MarlinSerial.cpp store_char() and MarlinSerial.h checkRx():
https://github.com/MarlinFirmware/Marlin/commit/a129078927c8968022569175cec7fea35c920fd2

Reply:
In theory we _could_ implement the real emergency parser, but it might not work very well. This is because it would make the UART ISR longer, taking up valuable clock cycles. Also, it’s important that this feature uses as little flash as possible since we are seriously running out of.

Reply:
btw I wonder what is taking so much space in prusa firmware? For example on my other printer (DIY corexy, with arduino mega + ramps 1.4) current Marlin 2 with tons of features (close to prusa ones but there are differences) turned on takes:
```
RAM:   [=======   ]  69.1% (used 5658 bytes from 8192 bytes)
Flash: [======    ]  62.7% (used 159308 bytes from 253952 bytes)
```

(if these PlatformIO stats are accurate)

Reply:
I think the UI is what kills most of the flash. The xyz calibration also takes a few percents. Translations are 10% of flash.
I think @wavexx knows more about this since he’s disabled significant parts of the firmware during testing before.

Reply:
Hard to say since not all parts can be easily disabled, but the whole ui/menu structure is dating back to marlin 1 and is very inefficient code-wise but would take some serious effort to improve around. xyzcal and selftest are relatively big. The current mmu code is scattered but it also takes up space, as well as multi-language support. Plus several small things that summed up I'm sure are eating space.. for example the la10->12 conversion doesn't take up much, but it's still extra stuff marlin2 doesn't need.